### ISSUE TITLE
How could we add the streaming support to enhance the output effect? 

### ISSUE THREAD

**Issue by Zephyruswind** [PromtEngineer/localGPT#546]
How could we add the streaming support to enhance the output effect?

**Comment by wangchao-sh** [comment]
I completely agree. The user experience is not satisfactory when the response is returned all at once. Is it possible to support a streaming approach where we can see the token output? It would greatly enhance the experience. It seems that the native Llama2 can achieve this, but I'm not sure how localGPT based on user data implements it.

**Comment by Rufus31415** [comment]
It's easy, just use ```callback_manager``` as a parameter for LlamaCPP and not in ```RetrievalQA``` and make llm verbose

See : https://github.com/Rufus31415/local-documents-gpt/blob/1f48f6884ae1d1fe474d6aa8e44b83480dfca30b/chat.py#L74C7-L74C7

```python
kwargs = {
#####
    "verbose": True
#####
}
callback_manager = CallbackManager([StreamingStdOutCallbackHandler()])
llm = LlamaCpp(callback_manager=callback_manager, **kwargs)
llm.streaming= True
```

**Comment by DrekoDev** [comment]
Would it be possible with Mistral ? @Rufus31415

**Comment by Rufus31415** [comment]
> Would it be possible with Mistral ? @Rufus31415

Yes, try this : https://huggingface.co/TheBloke/Mistral-7B-v0.1-GGUF

**Comment by Satyam7166-tech** [comment]
@Rufus31415 , will it hold for streamlit?

**Comment by Suiji12** [comment]
> > Would it be possible with Mistral ? @Rufus31415
> 
> Yes, try this : https://huggingface.co/TheBloke/Mistral-7B-v0.1-GGUF

Hello When I  run chat.py,I met a problem like this:
![1718118360640](https://github.com/PromtEngineer/localGPT/assets/113756481/b29532fa-c078-4cb2-9864-a555f118bc6d)

What can I do ?

**Comment by sorhtyre** [comment]
@Rufus31415, sorry to revive this old thread, but I was wondering the exact same thing. "How to implement streaming" I looked at the code snippet you provided and current source code, but I don't see how where this would go. Can you provide any further details?
